### PR TITLE
Bluetooth: Mesh: Add Option config unprov beacon interval

### PR DESCRIPTION
**Pull request by LingaoM**
1 Add Option to specifies the second interval when device
send Unprovisioned Beacon.

2 Move BT_MESH_LPN_SUB_ALL_NODES_ADDR to LPN subgroup.

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

### CODE REVIEW

File: subsys/bluetooth/mesh/Kconfig [hunk 1]
```diff
@@ -29,6 +29,18 @@ config BT_MESH_PB_ADV
 	  Enable this option to allow the device to be provisioned over
 	  the advertising bearer.
 
+if BT_MESH_PROV
+
+config BT_MESH_UNPROV_BEACON_INVL
+	int "The second interval to send the unallocated beacon"
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:35]:
```suggestion
	int "The second interval to send the unprovisioned beacon"
```

File: subsys/bluetooth/mesh/Kconfig [hunk 2]
```diff
@@ -29,6 +29,18 @@ config BT_MESH_PB_ADV
...
+	default 5
+	range 1 10
+	help
+	  This option specifies the second interval at which the
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:39]:
```suggestion
	  This option specifies the interval (in seconds) at which the
```

File: subsys/bluetooth/mesh/Kconfig [hunk 3]
```diff
@@ -29,6 +29,18 @@ config BT_MESH_PB_ADV
 	  Enable this option to allow the device to be provisioned over
 	  the advertising bearer.
 
+if BT_MESH_PROV
+
+config BT_MESH_UNPROV_BEACON_INVL
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/Kconfig:34]: Interval is generally shortened to `INT`, or the full `INTERVAL` is used.

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Just a couple of minor comments :)

**Review by trond-snekvik — APPROVED** [review]

**Comment by joerchan** [comment]
@LingaoM I think it would be best to have "2 Move BT_MESH_LPN_SUB_ALL_NODES_ADDR to LPN subgroup." as it's own commit.

**Comment by LingaoM** [comment]
> @LingaoM It would be best to have "2 Move BT_MESH_LPN_SUB_ALL_NODES_ADDR to LPN subgroup." as it's own commit.
> 
> Can you please split it?

Done

**Review by joerchan — APPROVED** [review]

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Please fix the compilation error before merging. When kconfig dependencies are missing, the integer config entries aren't defined at all, so you'll need to replace the `IS_ENABLED(CONFIG_BT_MESH_PROV)` with an `#if`

**Comment by joerchan** [comment]
@LingaoM Please check buildkite, there are compilation issues.

**Comment by LingaoM** [comment]
> @LingaoM Please check buildkite, there are compilation issues.

Done

File: subsys/bluetooth/mesh/beacon.c [hunk 4]
```diff
@@ -285,10 +288,13 @@ static void beacon_send(struct k_work *work)
 			k_delayed_work_submit(&beacon_timer,
 					      PROVISIONED_INTERVAL);
 		}
-	} else if (IS_ENABLED(CONFIG_BT_MESH_PB_ADV)) {
-		unprovisioned_beacon_send();
-		k_delayed_work_submit(&beacon_timer, UNPROVISIONED_INTERVAL);
+
+		return;
 	}
+#if defined(CONFIG_BT_MESH_PROV)
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/beacon.c:294]: Why is the dependency changed from BT_MESH_PB_ADV to BT_MESH_PROV?

**Review by jhedberg — CHANGES_REQUESTED** [review]
See my inline comments.

Also, your first commit message still contains stuff about BT_MESH_LPN_SUB_ALL_ADDRS even though you moved that to the second commit.

**Review by trond-snekvik — APPROVED** [review]

**Comment by jhedberg** [comment]
@LingaoM I pushed a third patch on top of your two patches, which eliminates a bunch of unnecessary #ifdefs from the code (in favour of using IS_ENABLED). Could you take a look and confirm that it's fine with you?

**Comment by LingaoM** [comment]
> @LingaoM I pushed a third patch on top of your two patches, which eliminates a bunch of unnecessary #ifdefs from the code (in favour of using IS_ENABLED). Could you take a look and confirm that it's fine with you?

LGTM

**Review by jhedberg — APPROVED** [review]

**Pull request merged**